hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/PreviewHandler.java

```diff
@@ -21,7 +24,8 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
         try {
             next.handleRequest(exchange);
         } finally {
-            messagePreviewLog.add(attachment.getTopic().getName(), attachment.getMessageContent());
+            byte[] previewBytes = converter.convert(attachment.getMessage());
+            messagePreviewLog.add(attachment.getTopic().getName(), previewBytes);
```

Comment:
this code runs for every message?

Reply:
i think it might be better to add this somewhere in log, so it runs for like 1 msg/minute instead for every msg

Reply:
I'll move it to the MessagePreviewFactory class, it seems it is a right place for it. Now it trims bytes to desired length and puts in MessagePreview object, it could convert whole Message object to Message preview instead of working on raw bytes.